### DISCUSSION TITLE
Feature Request: Detect Inactive Pucks

### DISCUSSION THREAD

**Discussion by Daleeburg** [RobertD502/home-assistant-flair#55] [Ideas]
Frist time HA user and know just enough coding to be dangerous.

The idea:
Detect when pucks go offline and allow it to trigger an action.  As it stands now, it does not appear that the "Inactive" data is passed through to HA's data set.

The data:
The Flair API does seem to track when a puck goes offline.  Calling "get https://api.flair.co/api/pucks" returns a bunch of data,  including "Data.attributes.inactive"

Inactive = "False" means the puck is online
Inactive = "True" means the puck is offline

The reason:
If my internet blips, one or both of the gateway pucks will go offline.  Following the Instructions for "Automatic Firmware Recovery" at https://support.flair.co/hc/en-us/articles/360040851932 seems to fix the problem.  I have configured them with Smart switches to be able to power cycle them remotely, but would love to just have HA detect and do it for me rather then coming home to a off temp house and having to go looking what failed.

Thoughts and am I doing this right?

**Comment by RobertD502** [comment]
You're in luck as this is already detected by the integration. You can use the Puck temp sensor, for example, to detect if a puck is "inactive". When a puck is offline, the sensor will become unavailable.

So, in your automation, check for when the state of the puck temp sensor entity is "unavailable" as your trigger. The same can be done for vent cover entities to determine if a vent is offline as reported by Flair's servers.

**Reply by Daleeburg** [reply]
I saw that in the API, but didnt know if that was reliable enough or if there were other things that would cause the "current-temperature-c" to return "Null".  Thank you very much for your wise council and I will set it up that way.

Thanks again for an awesome integration!

**Reply by RobertD502** [reply]
Even if a null value was returned for the temperature, it wouldn't affect the availability of the sensor entity - As you can see in the code [here](https://github.com/RobertD502/home-assistant-flair/blob/302e88ba4b83dc7aaeb45d62f97791b960d5ed82/custom_components/flair/sensor.py#L232-L239), I strictly check the inactive property of the puck to determine availability. So, a state of 'unavailable' will never be possible in the event that Flair's servers return a temp of null.

Hope this helps!

Edit: usage of the inactive property is the most reliable since the Flair servers will mark a puck inactive when a puck hasn't contacted the Flair servers in a while.
